### PR TITLE
Feat: improve payload parsing

### PR DESCRIPTION
**Pull request by Totodore**
This PR includes : 

- [x] better `Payload` parsing by directly returning a parsed packet to avoid any string allocation
- [x] fixing malicious length input from the user in v3
- [x] applying max_payload size to the `hyper::aggregate` fn to avoid server crash with malicious input
- [x] supporting utf8 packet parsing in v3

### CODE REVIEW

**Comment by Totodore** [comment]
So the solution I would like to implement consist of implementing the `Stream` trait for the `Payload` rather than `Iterator` and directly take the `http_body::Body` stream as input.

It would allow to buffer the input stream until a packet separator is found and then yield a new packet each time.

**Comment by biryukovmaxim** [comment]
> So the solution I would like to implement consist of implementing the `Stream` trait for the `Payload` rather than `Iterator` and directly take the `http_body::Body` stream as input.
> 
> It would allow to buffer the input stream until a packet separator is found and then yield a new packet each time.

Sounds really cool, like the solution

**Comment by Totodore** [comment]
So, I started to implement the streams wrapper for the request body in polling request version.
For V4 it is 100% ok.

However for V3 it is hard to optimize and do the less utf8 decoding and buffering as possible because of the char count based encoding.

Some issues remains for V3 and can be found with the `test_payload_stream_v3` test. It is a problem of a some after the first packet being buffered because we check the `packet_elen` in the chunk without taking in account the existing char length. However the existing char length decoded from the current packet buffer maybe bad because of some complex chars being splitted into different chunks and therefore having at the current iteration non utf8 data into the packet_buffer.

If anyone wan't to deep dive in this horror don't hesitate 😄.

Also everything deserve a refacto because it is currently really unreadable

**Comment by biryukovmaxim** [comment]
> If anyone wan't to deep dive in this horror don't hesitate smile.
Sure, that's about me


I created [pr](https://github.com/Totodore/socketioxide/pull/55
) fixing parsing
it still doesn't look good, but functionally it's okay

File: engineioxide/src/payload/mod.rs [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+                        packet_buf.truncate(i);
+                        packet_buf.len() - old_len
+                    } else {
+                        data.len()
+                    }
+                }
+            };
+            reader.consume(byte_read);
+            // Check if the packet length matches the number of characters
+            if let Ok(packet) = std::str::from_utf8(&packet_buf) {
```
Comment (biryukovmaxim) [inline engineioxide/src/payload/mod.rs:199]: `let Ok(packet) else { continue }`

**Pull request merged**